owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,365 @@
+        } returns Unit
+
+        ocLocalTransferDataSource.updateTransferWhenFinished(id, TransferStatus.TRANSFER_SUCCEEDED, timestamp, TransferResult.UPLOADED)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferWhenFinished(id, TransferStatus.TRANSFER_SUCCEEDED.value, timestamp, TransferResult.UPLOADED.value)
+        }
+    }
+
+    @Test
+    fun updateTransferLocalPath() {
+        every {
+            transferDao.updateTransferLocalPath(id, ocTransfer.localPath)
+        } returns Unit
+
+        ocLocalTransferDataSource.updateTransferLocalPath(id, ocTransfer.localPath)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferLocalPath(id, ocTransfer.localPath)
+        }
+    }
+
+    @Test
+    fun updateTransferStorageDirectoryInLocalPath() {
+        val oldDirectory = "oldDirectory"
+        val newDirectory = "newDirectory"
+
+        every {
+            transferDao.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+        } returns Unit
+
+        ocLocalTransferDataSource.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+        }
+    }
+
+    @Test
+    fun deleteTransferById() {
+        every {
+            transferDao.deleteTransferWithId(id)
+        } returns Unit
+
+        ocLocalTransferDataSource.deleteTransferById(id)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun deleteAllTransfersFromAccount() {
+        every {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        } returns Unit
+
+        ocLocalTransferDataSource.deleteAllTransfersFromAccount(OC_ACCOUNT_NAME)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        }
+    }
+
+    @Test
+    fun `getTransferById returns a OCTransfer`() {
+        every {
+            transferDao.getTransferWithId(any())
+        } returns ocTransfer.toEntity()
+
+        val actualResult = ocLocalTransferDataSource.getTransferById(id)
+
+        assertEquals(ocTransfer, actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun `getAllTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getAllTransfers()
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getAllTransfers()
+        }
+    }
+
+    @Test
+    fun `getAllTransfersAsStream returns a flow of list of OCTransfer with transferStatus in progress`() = runBlocking {
```

Comment:
Why `with transferStatus in progress`? That's not correct, we're retrieving transfers with the 4 possible status. Also, you should highlight here that they are retrieved ordered, which is the aim of the test:
```suggestion
    fun `getAllTransfersAsStream returns a flow of list of OCTransfer ordered by status`() = runBlocking {
```

Reply:
Done